src/Domain/ValueObjects/WalletValueObject.cs

```diff
@@ -0,0 +1,19 @@
+using Domain.Aggregates;
+namespace Domain.ValueObjects;
+
+public record Wallet(decimal Balance){
+    public User User { get; set; }
+    private decimal Balance { get; set; } = Balance;
```

Comment:
this property is not needed. C# records automatically generate these for you. for example:
```cs
record Person(Guid Id);

// generates a public initonly property
class Person
{
  public Guid Id { get; init; }

  public Person(Guid id)
  {
    Id = id;
  }
}
```

Reply:
i actually just needed it to implement negative number checking